### PR TITLE
Meta data for dataset, route to upload files and ID in nodes

### PR DESCRIPTION
**Pull request by Sinoeng**

### CODE REVIEW

File: Backend/src/db/setters.py [hunk 1]
```diff
@@ -26,3 +27,39 @@ def add_multiple_nodes_to(
...
     with ConnectionClass() as (_, db):
         db[collection].insert_many(nodes_to_insert)
+
+
+@validate_endpoint_args
+def add_multiple_nodes_to_id(
+    nodes: list[Node], collection: str, document_id: str, ConnectionClass=MongoConnection
+) -> None:
+    """
+    Bulk function to adding multiple nodes at once to a specific id.
```
Comment (Varlog22) [inline Backend/src/db/setters.py:37]: Rewrite suggestion:  Bulk function for adding multiple nodes at once to a specific id.

File: Backend/src/types/node.py [hunk 2]
```diff
@@ -6,11 +6,13 @@ class Node(BaseModel):
     Datamodel for a node
     """
 
-    id: str | None = None
+    id: str | None=No
```
Comment (Varlog22) [inline Backend/src/types/node.py:9]: should be id: str | None = None and the "cluster" field is not used in our new model so it should be removed.

File: Backend/src/v1/data.py [hunk 3]
```diff
@@ -39,9 +57,13 @@ def add_one_node(node: Node, collection: str) -> None:
 
 
 @router.post("/categorize", status_code=204)
-def categorize_node(node_id: str, category: str, colletion: str) -> None:
+def categorize_node(node_id: str, category: str, collection: str) -> None:
     """
     Categorize a node
     """
+    #TODO: not working
```
Comment (Varlog22) [inline Backend/src/v1/data.py:64]: if it is not working it should not be merged to main, keep it in a dev branch and work on it until it works.

**Review by Varlog22 — COMMENTED** [review]
Left some comments regarding the code, but the PR itself also needs to solve merge conflicts with main.

**Comment by jacomoel** [comment]
Think i have sorted out the problems that were due to poor branch management... PLEASE pull this branch before fixing anything PR comments requests!

File: .gitignore [hunk 4]
```diff
@@ -164,6 +164,9 @@ cython_debug/
 Backend/src/data/**/*.csv
 !Backend/src/data/.gitkeep
 
+Backend/src/data/info/**/.info
```
Comment (Leohemmingsson) [inline .gitignore:167]: This does not seem to work, the file is still there. I think you might mean to write `Backend/src/data/info/**/*.info`

**Review by Leohemmingsson — CHANGES_REQUESTED** [review]
This looks great! 

I just added a few comments I think could make the code a little cleaner

File: .gitignore (hunk 4 above)
Comment (jacomoel) [reply .gitignore]: done

File: Backend/src/v1/run_ml.py [hunk 5]
```diff
@@ -83,16 +90,17 @@ def run_ml_background_task(
...
+        about_dict = model_obj._read_meta_info(file_name=file)
+        model_obj.run(file_name=file, is_first=True, dim = dim_red_method)
     except Exception as e:
-        set_ml_status(
-            MlStatus(ml_id=ml_id, status="Not running", details=f"Error: {str(e)}")
-        )
-        raise e
+        raise HTTPException(status_code=400, detail=str(e))
+    
+    ref_id = f"{file}{"_id"}"
```
[the remark this answers is not in the document]
Comment (jacomoel) [reply Backend/src/v1/run_ml.py:98]: done

File: Backend/src/types/node.py [hunk 6]
```diff
@@ -21,6 +22,7 @@ class Node(BaseModel):
                     "x": 0.1,
                     "y": 0.2,
                     "truth": "true",
+                    "label": "true"
```
[the remark this answers is not in the document]
Comment (jacomoel) [reply Backend/src/types/node.py:25]: done

File: Backend/src/ml_lib/shared/shared_model.py [hunk 7]
```diff
@@ -64,3 +69,10 @@ def df(self) -> pd.DataFrame:
         Getter for the dataframe of ML models.
         """
         return self._df
+    
+    def get_id(self, content: str) -> str:
```
[the remark this answers is not in the document]
Comment (jacomoel) [reply Backend/src/ml_lib/shared/shared_model.py:73]: Yes @Sinoeng had the same idee about placing it in a util pkg, changed this to be declared in the shared_model for now, we might need to change it to util later.

File: Backend/src/ml_lib/shared/shared_model.py [hunk 8]
```diff
@@ -15,13 +14,21 @@ def _read_data_frame(self, file_name: str) -> pd.DataFrame:
...
         """
         Formatting logic, overridden by child classes.
         """
         pass
 
+    def _read_meta_info(self, file_name: str) -> dict:
+        """
+        Reading accompanying info file, overridden by child classes.
+        """
+        pass
```
[the remark this answers is not in the document]
Comment (jacomoel) [reply Backend/src/ml_lib/shared/shared_model.py:29]: change return type to -> dict | None

File: Backend/src/ml_lib/sub_models/qaqc_main_model.py [hunk 9]
```diff
@@ -41,6 +43,25 @@ def _format_data(self, file_name: str) -> None:
             writer.writerow(["text", "coarse label"])
             writer.writerows(zipped)
 
+    def _read_meta_info(self, file_name: str) -> dict:
```
[the remark this answers is not in the document]
Comment (jacomoel) [reply Backend/src/ml_lib/sub_models/qaqc_main_model.py:46]: done moved to filesystem

**Comment by jacomoel** [comment]
All the comments shall have been taken care of :)

File: .gitignore (hunk 4 above)
Comment (jacomoel) [reply .gitignore]: done

File: Backend/src/db/setters.py [hunk 10]
```diff
@@ -7,28 +7,48 @@
...
+            {"$push": {"data": {"$each": nodes_to_insert}}},
+            upsert=True
+        )
+
+@validate_endpoint_args
+def add_about_node_to_id(
+    about_node: dict, collection: str, document_id: str, ConnectionClass=MongoConnection
+) -> None:
+    """
+    Adds an about field with a document from a datasets accompanying .info file to a specific id.
```
[the remark this answers is not in the document]
Comment (jacomoel) [reply Backend/src/db/setters.py]: done

**Review by Leohemmingsson — APPROVED** [review]
Looks great, good job!

**Pull request merged**